grails5 upgrade

niravassar opened the pull request.

Commit:
grails5 upgrade

sdelamo commented on complete/build.gradle:
```diff
@@ -5,8 +5,8 @@ buildscript {
     }
     dependencies {
         classpath "org.grails:grails-gradle-plugin:$grailsVersion"
```
this should be decoupled from grails version

sdelamo commented on complete/build.gradle:
```diff
@@ -5,8 +5,8 @@ buildscript {
-        classpath "org.grails.plugins:hibernate5:${gormVersion-".RELEASE"}"
-        classpath "org.grails.plugins:views-gradle:1.2.0"
+        classpath "org.grails.plugins:hibernate5:7.1.0"
+        classpath "org.grails.plugins:views-gradle:2.0.0.RC2"
```
this should be 2.1.1

sdelamo commented on complete/gradle.properties:
```diff
@@ -1,3 +1,6 @@
-grailsVersion=3.3.0
-gormVersion=6.1.6.RELEASE
-gradleWrapperVersion=3.5
+grailsVersion=5.0.0
```
grails version should be 5.0.1

sdelamo commented on complete/gradle.properties:
```diff
@@ -1,3 +1,6 @@
-gormVersion=6.1.6.RELEASE
-gradleWrapperVersion=3.5
+grailsVersion=5.0.0
+gormVersion=7.1.0.M5
```
this should be GORM 7.1.0

sdelamo commented on complete/src/integration-test/groovy/intranet/backend/AnnouncementControllerSpec.groovy:
```diff
@@ -1,32 +1,36 @@
-            header("Accept-Version", "1.0") // <2>
-        }
+        HttpResponse<String> resp = client.toBlocking().exchange(request, String)
+        JSONArray jsonArray = new JSONArray(resp.body())
```
You can bind with Micronaut to a `List<Map>`

sdelamo commented on complete/src/integration-test/groovy/intranet/backend/AnnouncementControllerSpec.groovy:
```diff
@@ -35,21 +39,22 @@ class AnnouncementControllerSpec extends Specification {
-            header("Accept-Version", "2.0")
-        }
+        HttpResponse<String> resp = client.toBlocking().exchange(request, String)
+        JSONArray jsonArray = new JSONArray(resp.body())
```
You can bind with Micronaut to a `List<Map>`

sdelamo commented on complete/src/integration-test/groovy/intranet/backend/AnnouncementControllerSpec.groovy:
```diff
@@ -58,39 +63,41 @@ class AnnouncementControllerSpec extends Specification {
-            header("Accept-Version", "1.0")
-        }
+        HttpResponse<String> resp = client.toBlocking().exchange(request, String)
+        JSONObject jsonObject = new JSONObject(resp.body())
```
You can bind to a `Map` no need to bind to String and then to `JSONObject`

sdelamo commented on initial/build.gradle:
```diff
@@ -5,8 +5,8 @@ buildscript {
     }
     dependencies {
         classpath "org.grails:grails-gradle-plugin:$grailsVersion"
```
this should be decoupled from grails version

sdelamo commented on initial/build.gradle:
```diff
@@ -5,8 +5,8 @@ buildscript {
-        classpath "org.grails.plugins:hibernate5:${gormVersion-".RELEASE"}"
-        classpath "org.grails.plugins:views-gradle:1.2.0"
+        classpath "org.grails.plugins:hibernate5:7.1.0"
+        classpath "org.grails.plugins:views-gradle:2.0.0.RC2"
```
this should be 2.1.1

sdelamo commented on initial/gradle.properties:
```diff
@@ -1,3 +1,6 @@
-grailsVersion=3.3.0
-gormVersion=6.1.6.RELEASE
-gradleWrapperVersion=3.5
+grailsVersion=5.0.0
```
grails version should be 5.0.1

sdelamo commented on initial/gradle.properties:
```diff
@@ -1,3 +1,6 @@
-gormVersion=6.1.6.RELEASE
-gradleWrapperVersion=3.5
+grailsVersion=5.0.0
+gormVersion=7.1.0.M5
```
this should be GORM 7.1.0

sdelamo commented on initial/build.gradle:
```diff
@@ -25,41 +25,49 @@ repositories {
+    runtimeOnly "org.apache.tomcat:tomcat-jdbc"
+    testImplementation "org.grails:grails-gorm-testing-support"
+    testImplementation "org.grails.plugins:geb"
+    testImplementation "org.grails:grails-datastore-rest-client:6.1.9.RELEASE"
```
why do we need this dependency?

sdelamo commented on complete/src/integration-test/groovy/intranet/backend/AnnouncementControllerSpec.groovy:
```diff
@@ -58,39 +63,41 @@ class AnnouncementControllerSpec extends Specification {
+        request = HttpRequest.GET(UriBuilder.of("/announcements/${announcementId}").build())
+                .header("Accept-Version", "2.0");
+        resp = client.toBlocking().exchange(request, String)
+        jsonObject = new JSONObject(resp.body())
```
You can bind to a map or an object instead of String and then create a JSONObject

sdelamo commented on complete/build.gradle:
```diff
@@ -25,41 +25,49 @@ repositories {
+    runtimeOnly "org.apache.tomcat:tomcat-jdbc"
+    testImplementation "org.grails:grails-gorm-testing-support"
+    testImplementation "org.grails.plugins:geb"
+    testImplementation "org.grails:grails-datastore-rest-client:6.1.9.RELEASE"
```
why do we need this?

jamesfredley committed:
Update README to point at canonical guide on grails.apache.org (refs apache/grails-static-website#354)

jamesfredley committed:
Replace dead publisher CI with sample-app gradle check workflow

Delete: githubactions-build.sh
Add: .github/workflows/gradle.yml (auto-detects Java from Gradle wrapper version)

(refs apache/grails-static-website#354)
